source/geometry_model/box.cc

```diff
@@ -268,6 +269,115 @@ namespace aspect
       return std::min (std::max (d, 0.), maximal_depth());
     }
 
+    template <int dim>
+    void
+    Box<dim>::
+    update ()
+    {
+      this->get_pcout() << "   Updating surface values... " << std::endl;
+      TimerOutput::Scope timer_section(this->get_computing_timer(), "Geometry model surface update");
```

Comment:
Only do all of this if `dim == 2` for now, until we have an implementation for 3D as well.